pallets/ds-maps/src/lib.rs

```diff
@@ -308,9 +346,240 @@ impl Area {
+                    direction = left;
+                }
+            }
+
+            next_page_index = Self::get_index(current_cell_row_index - PAGE_LENGTH as u32, current_cell_column_index);
+            let (next_cell_row_index, _) = Self::extract_values_from_page_index(next_page_index);
+
+            if direction == left {
+                if next_cell_row_index >= first_cell_row_index {
+                    current_cell_row_index -= PAGE_LENGTH as u32;
+                    page_indexes.push(next_page_index);
+                    continue;
+                } else {
+                    direction = down;
+                }
+            }
+
+            next_page_index = Self::get_index(current_cell_row_index, current_cell_column_index - PAGE_WIDTH as u32);
+            let (_, next_cell_column_index) = Self::extract_values_from_page_index(next_page_index);
+            if direction == down {
+                if next_cell_column_index <= first_cell_column_index {
+                    current_cell_column_index -= PAGE_WIDTH as u32;
+                    page_indexes.push(next_page_index);
+                    continue;
+                } else {
+                    direction = right;
+                }
+            }
+        }
+
+        page_indexes
+    }
+
+    /// Gets the indexes of the cells where the point is located
+    pub fn get_cell_indexes(point: Point3D<Coord>) -> (u32, u32) {
+        let lat: u32 = point.lat.to_u32_with_frac_part(BITMAP_CELL_LENGTH, CELL_SIZE_DEGREE);
+        let lon: u32 = point.lon.to_u32_with_frac_part(BITMAP_CELL_WIDTH, CELL_SIZE_DEGREE);
+
+        let row_index: u32 = lat / BITMAP_CELL_LENGTH;
+        let column_index: u32 = lon / BITMAP_CELL_WIDTH;
+
+        (row_index, column_index)
+    }
+
+    /// Gets Page's index
+    pub fn get_index(cell_row_index: u32, cell_column_index: u32) -> u32 {
+        let row_index: u32;
+        let column_index: u32;
+
+        if cell_row_index > 0 && cell_row_index % PAGE_LENGTH as u32 != 0 {
+            row_index = PAGE_LENGTH as u32 + cell_row_index - cell_row_index % PAGE_LENGTH as u32;
+        } else if cell_row_index == 0 {
+            row_index = PAGE_LENGTH as u32;
+        } else {
+            row_index = cell_row_index;
+        }
+
+        if cell_column_index > 0 && cell_column_index % PAGE_WIDTH as u32 != 0 {
+            column_index = PAGE_WIDTH as u32 + cell_column_index - cell_column_index % PAGE_WIDTH as u32;
+        } else if cell_column_index == 0 {
+            column_index = PAGE_WIDTH as u32;
+        } else {
+            column_index = cell_column_index;
+        }
+
+        (row_index << 16) | column_index
+    }
+
+    /// Gets boundary cells (southwest and northeast) indexes from Page index.
+    /// Returns the row and column of the southwest cell and the row and column of the northeast
+    /// cell, respectively.
+    pub fn get_boundary_cell_indexes(
+        index: u32, sw_cell_row_index: u32, ne_cell_column_index: u32,
```

Comment:
If it's number of Row+Column, used throughout all pallet, maybe better give it own type(like RootId etc) for readability?

Reply:
About the return values: according to the structure of the Page index, only two numbers are stored there (storing 4 is redundant), while 4 numbers are needed for other calculations when using boundary cells.